File: wordle/wordle.py
import random
import os
import sys 
# from colored import fg, attr
FgGray = "\x1b[90m"
FgYellow = "\x1b[33m"
FgGreen = "\x1b[32m"
Reset = "\x1b[0m"
# board = []
def get_word_list():
    # Create a list of words for the game; the more the merrier!

    # YOUR CODE GOES HERE
    return ['tasco', 'scalp', 'tress', 'folks']
    pass

def pick_random_word(word_list):
    # Use the random module to pick a random word from the list
    # YOUR CODE GOES HERE
    return random.choice(word_list)
    pass

def color_letter(letter, color):
    # Use the colored module to color the letter

    # YOUR CODE GOES HERE
    return f'{color}{letter}{Reset}'
    pass

def evaluate_guess(guess, secret_word):
    # Determine how many letters in the guess are in the secret word
    words = []
    # YOUR CODE GOES HERE
    for i in range(len(guess)):
        letter = guess[i]
        lIndex = secret_word.find(letter)
        if lIndex >= 0:
            if lIndex == guess.find(letter):
                words.append(color_letter(letter=letter,color=FgGreen))
            else:
                words.append(color_letter(letter=letter,color=FgYellow))
        else:
            words.append(color_letter(letter=letter, color=FgGray))
        pass 
    return words;
    pass
# global index_of_tries
# index_of_tries = 0
def play_wordle():
    # Loop through the game until the user guesses the word or runs out of guesses

    # YOUR CODE GOES HERE
    
    print("Starting Wordle...")
    word = pick_random_word(get_word_list())
    index = 0
    board = []
    def run(err, index):
        # global index
        if os.environ.get('REVAL_WORD'):
            print(f'Cheater: {word}')
        print(f'Try {index}/6')
        print('============================================')
        if len(board) > 0:
            for row in board:
                print(''.join(row))
        else:
            print('<empty board>')
        print('============================================')
        if err:
            print(err)
        # print()
        inp = input('Enter your guess: ')
        if inp.isalpha() == False or len(inp) < 5:
            os.system('cls||clear')
            run(err="Make sure you only have letters and have 5 chars", index=index)
            return;
        inp = inp[0:5]
        nextRow = evaluate_guess(inp, word)
        if len(board) >= 5:
            if inp == word:
                print(f'{FgGreen}You Won?{Reset}')
            else:
                print(f'{FgGray}You lost{Reset}')
            quit()
        else:
            if inp == word:
                print(f'{FgGreen}You Won?{Reset}')
                quit()
            else:
                board.append(nextRow)
                os.system('cls||clear')
                index += 1
                run(err=None,index=index)
        pass
    pass
    run(err=None,index=index)

if __name__ == "__main__":
    play_wordle()

File: wordle/wordle.js
Reset = "\x1b[0m"
Bright = "\x1b[1m"
Dim = "\x1b[2m"
Underscore = "\x1b[4m"
Blink = "\x1b[5m"
Reverse = "\x1b[7m"
Hidden = "\x1b[8m"

FgBlack = "\x1b[30m"
FgRed = "\x1b[31m"
FgGreen = "\x1b[32m"
FgYellow = "\x1b[33m"
FgBlue = "\x1b[34m"
FgMagenta = "\x1b[35m"
FgCyan = "\x1b[36m"
FgWhite = "\x1b[37m"
FgGray = "\x1b[90m"

BgBlack = "\x1b[40m"
BgRed = "\x1b[41m"
BgGreen = "\x1b[42m"
BgYellow = "\x1b[43m"
BgBlue = "\x1b[44m"
BgMagenta = "\x1b[45m"
BgCyan = "\x1b[46m"
BgWhite = "\x1b[47m"
BgGray = "\x1b[100m"
if(!process.env.TERM.includes('kitty')) {
    console.log(`Use Kitty Terminal app for better preformance`)
    process.exit(1)
}
const readLine = () => new Promise((r) => {
    process.stdin.once('data', d => r(d.toString()))
})

// mimic py functions
const get_word_list = () => {
    // for js one im doing random stuff
    return ['Tasco', 'Scalp', 'Tress', 'Folks'].map(w => w.toLowerCase().trim()).sort((a,b) => Math.random() > .5 ? 1 -2 : 2 - 1)
}
const pick_random_word = (list) => list[Math.floor(Math.random() * list.length)]

const color_letter = (letter, color) => `${color}${letter}${Reset}`
const evaluate_guess = (guess, secret_word) => {
    const words = []
    //Determine how many letters in the guess are in the secret word
    for(const letter of guess.split('')) {
     const lIndex = secret_word.indexOf(letter)
     if(lIndex >= 0) {
if(lIndex == guess.indexOf(letter)) {
    words.push(color_letter(letter, FgGreen))
} else {
    words.push(color_letter(letter, FgYellow))
}
     } else {
        words.push(color_letter(letter, FgGray))
     }
       
    }
    return words;
}
 function play_wordle() {
    console.log(`Starting wordle...`)
    const word = pick_random_word(get_word_list())
    if(process.env.REVEAL_WORD) console.log(`Cheater: ${word}`)
    // console.log(`Your wordle input`)
    let board = []
async function run(err) {
    console.log(new Array(40).join('='))
if(board.length > 0) {
    board.forEach(row => console.log(row.join(' ')))
} else {
    console.log('<empty board>')
}
    console.log(new Array(40).join('='))
    if(err) console.warn(err)
    process.stdout.write(`Enter your guess: `)
    const input = await readLine().then(str => str.replace(/[^a-zA-Z]+/g, '').slice(0, 5))
    if(input.length < 5) {
        // console.log(``)
        console.clear()
        return run("Make sure your guess is 5 Characters!");
    }
    const newRow =  evaluate_guess(input, word)
    if(board.length >= 5) {
        // console.log(board[board.length - 1])
        // const maybeWord = board[board.length - 1].join('').replaceAll(Reset, '').replaceAll(FgYellow, '').replaceAll(FgGray, '').replaceAll(FgGreen, '').trim()
        // console.log(`Does '${maybeWord}' == '${word}'`)
if(input == word) {
    console.log(`${FgGreen}You won??${Reset} `)
    setTimeout(() => process.exit(1), 50)
    return;
} else {
    console.log(`${FgRed} You Lost! ${Reset}`)
    setTimeout(() => process.exit(1), 50)
    return;
}
        return;
    } else {
    if(input == word) {
    console.log(`${FgGreen}You won??${Reset} `)
        // console.log(`You won??`)
        setTimeout(() => process.exit(1), 50)
        return;
    } else {
        board.push(newRow)
        console.clear()
        run()
    }
    }
}
run()
}
play_wordle()